### ISSUE TITLE
"react/prop-types" array length with optional chaining

### ISSUE THREAD

**Issue by ops1ops** [jsx-eslint/eslint-plugin-react#3309]
Hello, when i use optional chaining with array (`array?.length`), react/prop-types warns that length is missing in props validation, but if i use `(array || []).length` it does not warn

Type used for array: `array?: Array<Entity>`

Exact error: `ESLint: 'array.length' is missing in props validation(react/prop-types)`

Environment:
```
"typescript": "^3.7.5",
"eslint": "^7.1.0",
"eslint-plugin-react": "^7.30.0",
```

**Comment by ljharb** [comment]
That’s because it’s an optional type - it could be undefined.

**Comment by ops1ops** [comment]
No, the problem was that i had cached lower version of library (about 7.10), i reinstalled node modules with 7.30 and now it does not warn, seems it was fixed between 10 and 30 versions, thats why i closed the issue
